.github: Consolidate binary checkout logic

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73092

Consolidates binary checkout logic to use the standard common logic we
have in our common templates. Also fixes issues related to
pytorch/builder trying to checkout the head commit for pytorch/pytorch
instead of checking out the builder commit we actually want

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
.github: Consolidate binary checkout logic

Consolidates binary checkout logic to use the standard common logic we
have in our common templates. Also fixes issues related to
pytorch/builder trying to checkout the head commit for pytorch/pytorch
instead of checking out the builder commit we actually want

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

malfet commented on .github/templates/common.yml.j2:
```diff
@@ -6,6 +6,10 @@
 
+# NOTE: If testing pytorch/builder changes you can change this variable to change what pytorch/builder reference
+#       the binary builds will check out
+{%- set builder_branch = "master" -%}
```
It should be main, shouldn't it?
```suggestion
{%- set builder_branch = "main" -%}
```

seemethere replied:
Fixed

seemethere committed:
Update on ".github: Consolidate binary checkout logic"


Consolidates binary checkout logic to use the standard common logic we
have in our common templates. Also fixes issues related to
pytorch/builder trying to checkout the head commit for pytorch/pytorch
instead of checking out the builder commit we actually want

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented:
@pytorchbot merge this
